homeassistant/components/websocket_api/commands.py

```diff
@@ -221,22 +223,25 @@ async def handle_call_service(
     if template.is_complex(target):
         raise vol.Invalid("Templates are not supported here")
 
+    domain = msg["domain"]
+    service_name = msg["service"]
+    all_services = hass.services.async_services()
+    if (service := all_services.get(domain, {}).get(service_name)) is None:
```

Comment:
Why do we change how ServiceNotFound is handled?

Reply:
because we can't determine whether or not the service supports responses without accessing the `Service` instance. In order to access the instance, I have to assume it may not exist, so rather than duplicate the check, I just moved it to here